homeassistant/components/fritzbox/climate.py

```diff
@@ -88,6 +88,8 @@ def precision(self) -> float:
     @property
     def current_temperature(self) -> float:
         """Return the current temperature."""
+        if self.device.has_temperature_sensor and self.device.temperature is not None:
+            return self.device.temperature  # type: ignore [no-any-return]
         return self.device.actual_temperature  # type: ignore [no-any-return]
```

Comment:
Does this even happen? I think a device that `has_temperature_sensor` will always have the `temperature` attribute, but only a device that is also a thermostat will additionally have the `actual_temperature` attribute.

Reply:
TBH i agree with you - a thermostat has usual always also a temperature sensor, but i left this in as a guard 🤔